numba/tests/test_function.py

```diff
@@ -0,0 +1,575 @@
+            return 3
+
+        def g(arg):
+            if arg:
+                f = a
+            else:
+                f = b
+            out = f()
+            return out
+
+        self.assertEqual(njit(g)(True), 2)
+        self.assertEqual(njit(g)(False), 3)
+
+    def test_pr4967_example(self):
+
+        @cfunc('int64(int64)')
+        def a(i):
+            return i + 1
+
+        @cfunc('int64(int64)')
+        def b(i):
+            return i + 2
+
+        @njit
+        def foo(f, g):
+            i = f(2)
+            seq = (f, g)
+            for fun in seq:
+                i += fun(i)
+            return i
+
+        a_ = a._pyfunc
+        b_ = b._pyfunc
+        self.assertEqual(foo(a, b),
+                         a_(2) + a_(a_(2)) + b_(a_(2) + a_(a_(2))))
+
+    def test_in(self):
+
+        def a(i):
+            return i + 1
+
+        def foo(f):
+            return 0
+
+        for decor in all_func_kinds:
+            if decor in [pure_func]:
+                # Skip not-yet-supported functions
+                continue
+            with self.subTest(decor=decor.__name__):
+                a_ = decor(a)
+                self.assertEqual(njit(foo)(a_), foo(a))
+
+    def test_in_call(self):
+
+        def a(i):
+            return i + 1
+
+        def foo(f):
+            r = f(123)
+            return r
+
+        for decor in all_func_kinds:
+            if decor in [pure_func, ctypes_func]:
+                # Skip not-yet-supported functions
+                continue
+            with self.subTest(decor=decor.__name__):
+                a_ = decor(a)
+                self.assertEqual(njit(foo)(a_), foo(a))
+
+    def test_in_call_out(self):
+
+        def a(i):
+            return i + 1
+
+        def foo(f):
+            f(123)
+            return f
+
+        for decor in all_func_kinds:
+            if decor in [pure_func, ctypes_func, njit_func, njit2_func,
+                         wap_func]:
+                # Skip not-yet-supported functions
+                continue
+            with self.subTest(decor=decor.__name__):
+                a_ = decor(a)
+                if decor is cfunc_func:
+                    self.assertEqual(njit(foo)(a_).pyfunc, foo(a))
+                else:
+                    self.assertEqual(njit(foo)(a_), foo(a))
+
+    def test_in_seq_call(self):
+
+        def a(i):
+            return i + 1
+
+        def b(i):
+            return i + 2
+
+        def foo(f, g):
+            r = 0
+            for f_ in (f, g):
+                r = r + f_(r)
+            return r
+
+        for decor in all_func_kinds:
+            if decor in [pure_func, ctypes_func, njit_func, njit2_func]:
+                # Skip not-yet-supported functions
+                continue
+            with self.subTest(decor=decor.__name__):
+                a_ = decor(a)
+                b_ = decor(b)
+                self.assertEqual(njit(foo)(a_, b_), foo(a, b))
+
+    def test_in_ns_seq_call(self):
```

Comment:
I feel that this test name needs expanding, or a doc string for what it is testing.

Reply:
Added doc string.